src/Build/Evaluation/Evaluator.cs

```diff
@@ -2375,6 +2375,29 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
             if (!EvaluateConditionCollectingConditionedProperties(importElement, ExpanderOptions.ExpandProperties,
                 ParserOptions.AllowProperties, _projectRootElementCache))
             {
+                if (!_evaluationLoggingContext.LoggingService.OnlyLogCriticalEvents)
```

Comment:
I think I changed my mind about having an escape hatch for this since yesterday. It's still not good enough if this causes an RPS regression, but it could be useful if it causes problems in some obscure scenario for a specific customer.